oracledb-ballerina/client.bal

```diff
@@ -0,0 +1,213 @@
+    #            remaining commands in the batch after a failure. The summary of the executed queries in case of error
+    #            can be accessed as `(<sql:BatchExecuteError> result).detail()?.executionResults`
+    remote isolated function batchExecute(@untainted sql:ParameterizedQuery[] sqlQueries)
+    returns sql:ExecutionResult[]|sql:Error {
+        if (sqlQueries.length() == 0) {
+            return error sql:ApplicationError(" Parameter 'sqlQueries' cannot be empty array");
+        }
+        if (self.clientActive) {
+            return nativeBatchExecute(self, sqlQueries);
+        } else {
+            return error sql:ApplicationError(
+                "OracleDB client is already closed, hence further operations are not allowed");
+        }
+    }
+
+    # Executes a SQL stored procedure and returns the result as stream and execution summary.
+    #
+    # + sqlQuery - The query to execute the SQL stored procedure
+    # + rowTypes - The array of `typedesc` of the records that should be returned as a result. If this is not provided
+    #              the default column names of the query result set be used for the record attributes
+    # + return - Summary of the execution is returned in `ProcedureCallResult` or `sql:Error`
+    remote isolated function call(@untainted string|sql:ParameterizedCallQuery sqlQuery, typedesc<record {}>[] rowTypes = [])
+    returns sql:ProcedureCallResult|sql:Error {
+        if (self.clientActive) {
+            return nativeCall(self, sqlQuery, rowTypes);
+        } else {
+            return error sql:ApplicationError("OracleDB client is already closed, hence further operations are not allowed");
+        }
+    }
+
+    # Close the SQL client.
+    #
+    # + return - Possible error during closing the client
+    public isolated function close() returns sql:Error? {
+        self.clientActive = false;
+        return close(self);
+    }
+
+}
+
+# SSL Configuration to be used when connecting to oracle server.
+#
+# + keyStore - Keystore configuration of the client certificates
+# + trustStore - Truststore configuration of the trust certificates
+# + keyStoreType - The type of the keystore - "JKS" / "PKCS12" / "SSO"
+# + trustStoreType - The type of the keystore - "JKS" / "PKCS12" / "SSO"
+public type SSLConfig record {|
+  crypto:KeyStore keyStore?;
+  crypto:TrustStore trustStore?;
+  string keyStoreType?;
+  string trustStoreType?;
+|};
+
+# Oracle database specific JDBC options.
+#
+# + ssl - SSL Configuration to be used
+# + loginTimeout - Specify how long to wait for establishment of a database connection in seconds
+# + autoCommit - If true commits automatically when statement is complete
+# + connectTimeout - Time duration for a connection
+# + socketTimeout - Timeout duration for reading from a socket
+public type Options record {|
+   SSLConfig ssl?;
+   decimal loginTimeout?;
+   boolean autoCommit?;
+   decimal connectTimeout?;
+   decimal socketTimeout?;
```

Comment:
is there a default value for these options, if so, better if we can reflect them here

Reply:
Resolved with a2696e1433d2aa6528411382dd4340e4357a845c